Source/engine/game_assets.cpp

```diff
@@ -0,0 +1,76 @@
+#include "engine/game_assets.hpp"
+
+#include <algorithm>
+#include <cstdint>
+#include <cstring>
+
+#include "init.h"
+#include "utils/file_util.h"
+#include "utils/log.hpp"
+#include "utils/mpq_sdl_rwops.hpp"
+#include "utils/paths.h"
+
+namespace devilution {
+
+namespace {
+
+bool OpenMpqFile(const char *filename, MpqArchive **archive, uint32_t *fileNumber)
+{
+	const MpqArchive::FileHash fileHash = MpqArchive::CalculateFileHash(filename);
+	const auto at = [=](std::optional<MpqArchive> &src) -> bool {
+		if (src && src->GetFileNumber(fileHash, *fileNumber)) {
+			*archive = &(*src);
+			return true;
+		}
+		return false;
+	};
+
+	return at(font_mpq) || at(lang_mpq) || at(devilutionx_mpq)
+	    || (gbIsHellfire && (at(hfvoice_mpq) || at(hfmusic_mpq) || at(hfbarb_mpq) || at(hfbard_mpq) || at(hfmonk_mpq) || at(hellfire_mpq))) || at(spawn_mpq) || at(diabdat_mpq);
+}
+
+} // namespace
+
+SDL_RWops *OpenGameAsset(const char *filename, bool threadsafe)

```

Comment:
For some reason I prefer to not have `Game` in this name, it some how seams either redundant or too specific: `OpenAsset()`, what do you think?

Reply:
Renamed to OpenAsset